packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -5723,6 +5763,185 @@ class _ScribbleFocusableState extends State<_ScribbleFocusable> implements Scrib
+      rect.bottom + padding.bottom,
+    );
+  }
+
+  /// Given a [Rect] in a [RenderBox]'s local coordinate space, returns that
+  /// [Rect] in global coordinates.
+  static Rect _localToGlobalRect(Rect rect, RenderBox renderBox) {
+    return Rect.fromPoints(
+      renderBox.localToGlobal(rect.topLeft),
+      renderBox.localToGlobal(rect.bottomRight),
+    );
+  }
+
+  Rect? _getHandleRect(TextSelectionHandleType type) {
+    if (widget.selectionControls == null) {
+      return null;
+    }
+    // Do not expand the Rect to kMinInteractiveDimension because it will
+    // targeted by a precise pointing device.
+    return widget.selectionControls!.getHandleRect(type, _renderEditable.preferredLineHeight);
+  }
+
+  RenderEditable get _renderEditable => widget.editableKey.currentContext!.findRenderObject()! as RenderEditable;
+
+  Future<void> _handlePointerEvent(PointerEvent event) async {
+    if (event is! PointerDownEvent
+      || event.kind != ui.PointerDeviceKind.stylus
+      || !(await Scribe.isFeatureAvailable() ?? false)) {
+      return;
+    }
+
+    final RenderBox renderBox = widget.editableKey.currentContext!.findRenderObject()! as RenderBox;
+
+    final Rect renderBoxRect = _localToGlobalRect(renderBox.paintBounds, renderBox);
+    final Rect hitRect = _pad(renderBoxRect, _handwritingPadding);
+    if (!hitRect.contains(event.position)) {
+      return;
+    }
+
+    // A stylus event that starts on a selection handle does not start
+    // handwriting, it moves the handle.
+    final Offset? startHandleOffset = _renderEditable.startHandleLayerLink.leader?.offset;
+    if (startHandleOffset != null) {
+      final Rect? leftHandleRectLocal = _getHandleRect(TextSelectionHandleType.left);
+      if (leftHandleRectLocal != null && !leftHandleRectLocal.isEmpty) {
+        final Rect leftHandleRectGlobal = _localToGlobalRect(leftHandleRectLocal, renderBox);
+        final Rect leftHandleRect = leftHandleRectGlobal.shift(startHandleOffset);
+        if (leftHandleRect.contains(event.position)) {
+          return;
+        }
+      }
+    }
+    final Offset? endHandleOffset = _renderEditable.endHandleLayerLink.leader?.offset;
+    if (endHandleOffset != null) {
+      final Rect? rightHandleRectLocal = _getHandleRect(TextSelectionHandleType.right);
+      if (rightHandleRectLocal != null && !rightHandleRectLocal.isEmpty) {
+        final Rect rightHandleRectGlobal = _localToGlobalRect(rightHandleRectLocal, renderBox);
+        final Rect rightHandleRect = rightHandleRectGlobal.shift(endHandleOffset);
+        if (rightHandleRect.contains(event.position)) {
+          return;
+        }
+      }
+    }
+
+    if (!widget.focusNode.hasFocus) {
+      widget.focusNode.requestFocus();
+    }
+
+    return Scribe.startStylusHandwriting();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.enabled) {
+      GestureBinding.instance.pointerRouter.addGlobalRoute(_handlePointerEvent);
```

Comment:
Also why can't this be a regular gesture recognizer / detector? Is it because of the padding?

Reply:
I will follow up on this one on Monday.

 > What prevents 2 overlapping text fields from competing for stylus point down events?

Good catch, I just tried this out and it looks like the behavior to the user is that the field lowest in the tree will win if I write directly in between the two fields, where their paddings are overlapping. If I get really accurate with the stylus I can write just in the bottom of the upper field but the lower field will receive the text, but for the most part the behavior is probably fine in this ambiguous situation.

However, both fields do receive the event. I'll need to follow up and do something to prevent that.

<details>

<summary>3 TextFields right next to each other</summary>

```dart
child: Column(
  mainAxisAlignment: MainAxisAlignment.center,
  children: <Widget>[
    TextField(
      controller: controller,
      stylusHandwritingEnabled: true,
      decoration: InputDecoration(
        border: OutlineInputBorder(),
      ),
    ),
    TextField(
      stylusHandwritingEnabled: true,
      decoration: InputDecoration(
        border: OutlineInputBorder(),
      ),
    ),
    TextField(
      stylusHandwritingEnabled: true,
      decoration: InputDecoration(
        border: OutlineInputBorder(),
      ),
    ),
  ],
),
```

</details>

 > Also, the stylus hitting an invisible text field (in an inactive route, for example) would also trigger scribble it seems?

Should I check `ModalRoute.of(context)` or something like that? Anything else I should check?

 > Also why can't this be a regular gesture recognizer / detector? Is it because of the padding?

Yes, this was my solution to needing to listen to an area outside of the size of the field.